File: Dynamic_Programming/21_Oct_Eulerian_number/21_Oct_Eulerian_number.py
#naive eulerian

def eulerian(n,m):
    if m >= n or n == 0:
        return 0
    if m == 0:
        return 1
    
    return ((n-m)*eulerian(n-1,m-1))+((m+1)*eulerian(n-1,m))


#driver code

n = 3
m = 1
print("Eulerian(",n,",", m, ") is ", eulerian(n,m))

File: Dynamic_Programming/october/21_Oct_Eulerian_number/21_Oct_Eulerian_number.cpp
//naive eulerian number

#include<bits/stdc++.h>

using namespace std;

int eulerian(int n, int m){
    if(m >= n || n == 0){
        return 0;
    }
    if( m == 0){
        return 1;
    }

    return (n-m)*eulerian(n-1,m-1) + 
            (m+1)*eulerian(n-1,m);
}

//driver code
int main(){
    int n = 3, m = 1;

    cout << "Eulerian (" << n << "," << m<< ") is " << eulerian(n,m);

    return 0;
}